frame/vamm/src/tests/swap.rs

```diff
@@ -332,13 +332,40 @@ fn should_not_update_twap_if_current_twap_timestamp_is_more_recent() {
 		assert_ne!(vamm_state_t0.twap_timestamp, vamm_state_t1.twap_timestamp);
 
 		// In the second swap the time didn't pass between operations, so we
-		// can't allow the twap value to be updated again.
+		// can't allow the twap timestamp nor the twap value to be updated
+		// again.
 		assert_ok!(TestPallet::swap(&swap_config));
 		let vamm_state_t2 = TestPallet::get_vamm(0).unwrap();
 		assert_eq!(vamm_state_t1.twap_timestamp, vamm_state_t2.twap_timestamp);
+		assert_eq!(vamm_state_t1.base_asset_twap, vamm_state_t2.base_asset_twap);
 	});
 }
 
+#[rstest]
+#[case(Timestamp::MIN)]
+#[case(0)]
+#[case(1)]
+#[case(10)]
+#[case(100)]
+#[case(1000)]
+#[case(10000)]
+fn should_update_twap_if_twap_period_has_passed(#[case] delta: Timestamp) {
+	with_swap_context(
+		TestVammConfig { base_asset_reserves: as_decimal_inner(42), ..Default::default() },
+		TestSwapConfig { input_amount: as_decimal_inner(30), ..Default::default() },
+		|_, swap_config| {
+			let vamm_state_t0 = TestPallet::get_vamm(0).unwrap();
+			run_for_seconds(delta.saturating_add(vamm_state_t0.twap_period));
+			assert_ok!(TestPallet::swap(&swap_config));
+			run_for_seconds(delta.saturating_add(vamm_state_t0.twap_period));
+			assert_ok!(TestPallet::swap(&swap_config));
```

Comment:
Why does it have to a delta **plus** the 'twap period'? Again, I think the naming might be causing confusion here, `twap_period` is only used to calculate the weights in the weighted average between the current price and last TWAP, not as a reference for how much time should pass between swaps

Reply:
Unless this test is supposed to check if the TWAP is updated *even if* more than 'twap_period' has passed.

Reply:
My last question would be: why do 2 swaps here if you don't get the vAMM state in between them?